src/lib/ngsi/ContextElementResponse.cpp

```diff
@@ -125,7 +125,119 @@ ContextElementResponse::ContextElementResponse
+    }
+    else
+    {
+      switch(getFieldF(attr, ENT_ATTRS_VALUE).type())
+      {
+      case orion::String:
+        ca.stringValue = getStringFieldF(attr, ENT_ATTRS_VALUE);
+        if (!includeEmpty && ca.stringValue.empty())
+        {
+          continue;
+        }
+        caP = new ContextAttribute(ca.name, ca.type, ca.stringValue);
+        break;
+
+      case orion::NumberDouble:
+        ca.numberValue = getNumberFieldF(attr, ENT_ATTRS_VALUE);
+        caP = new ContextAttribute(ca.name, ca.type, ca.numberValue);
+        break;
+
+      case orion::NumberInt:
+        ca.numberValue = (double) getIntFieldF(attr, ENT_ATTRS_VALUE);
+        caP = new ContextAttribute(ca.name, ca.type, ca.numberValue);
+        break;
+
+      case orion::Bool:
+        ca.boolValue = getBoolFieldF(attr, ENT_ATTRS_VALUE);
+        caP = new ContextAttribute(ca.name, ca.type, ca.boolValue);
+        break;
+
+      case orion::jstNULL:
+        caP = new ContextAttribute(ca.name, ca.type, "");
+        caP->valueType = orion::ValueTypeNull;
+        break;
+
+      case orion::Object:
+        caP = new ContextAttribute(ca.name, ca.type, "");
+        caP->compoundValueP = new orion::CompoundValueNode(orion::ValueTypeObject);
+        caP->valueType = orion::ValueTypeObject;
+        compoundObjectResponse(caP->compoundValueP, getFieldF(attr, ENT_ATTRS_VALUE));
+        break;
+
+      case orion::Array:
+        caP = new ContextAttribute(ca.name, ca.type, "");
+        caP->compoundValueP = new orion::CompoundValueNode(orion::ValueTypeVector);
+        caP->valueType = orion::ValueTypeVector;
+        compoundVectorResponse(caP->compoundValueP, getFieldF(attr, ENT_ATTRS_VALUE));
+        break;
+
+      default:
+        LM_E(("Runtime Error (unknown attribute value type in DB: %d)", getFieldF(attr, ENT_ATTRS_VALUE).type()));
+      }
+    }
+
+    /* dateExpires is managed like a regular attribute in DB, but it is a builtin and it is shadowed */
+    if (caP->name == DATE_EXPIRES)
+    {
+      caP->shadowed = true;
+    }
+
+    /* Setting custom metadata (if any) */
+    if (attr.hasField(ENT_ATTRS_MD))
+    {
+      orion::BSONObj         mds = getObjectFieldF(attr, ENT_ATTRS_MD);
+      std::set<std::string>  mdsSet;
+
+      mds.getFieldNames(&mdsSet);
+      for (std::set<std::string>::iterator i = mdsSet.begin(); i != mdsSet.end(); ++i)
+      {
+        std::string currentMd = *i;
+        Metadata*   md = new Metadata(dbDecode(currentMd), getObjectFieldF(mds, currentMd));
+        caP->metadataVector.push_back(md);
+      }
+    }
+
+    /* Set creDate and modDate at attribute level */
+    if (attr.hasField(ENT_ATTRS_CREATION_DATE))
+    {
+      caP->creDate = getNumberFieldF(attr, ENT_ATTRS_CREATION_DATE);
+    }
+
+    if (attr.hasField(ENT_ATTRS_MODIFICATION_DATE))
+    {
+      caP->modDate = getNumberFieldF(attr, ENT_ATTRS_MODIFICATION_DATE);
+    }
+
+    entity.attributeVector.push_back(caP);
+  }
```

Comment:
This seems to be a big block of duplicated code. The changes done PR https://github.com/telefonicaid/fiware-orion/pull/4491 confirm and fix it.

NTC (post-merge comment)